File: AAS.Domain/Users/Roles/UserRole.cs
#region

using AAS.Domain.AccessPolicies;
using AAS.Tools.Types.IDs;

#endregion

namespace AAS.Domain.Users.Roles;

public class UserRole
{
    public ID Id { get; }
    public string Name { get; }
    public AccessPolicy[] AccessPolicies { get; }

    public UserRole(ID id, string name, AccessPolicy[] accessPolicies)
    {
        Id = id;
        Name = name;
        AccessPolicies = accessPolicies;
    }
}

File: AAS.BackOffice/client_app/src/domain/users/roles/userRole.ts
import {AccessPolicy} from "../../accessPolicies/accessPolicy";

export class UserRole {
    constructor(
        public readonly id: string,
        public readonly name: string,
        public readonly accessPolicies: AccessPolicy[]
    ) {
    }
}

export function mapToUserRole(value: any): UserRole {
    return new UserRole(value.id, value.name, value.accessPolicies);
}

export function mapToUserRoles(values: any[]): UserRole[] {
    return values.map(mapToUserRole);
}